content/developer/reference/frontend/owl_components.rst

```diff
@@ -150,13 +150,144 @@ checkboxes or datepickers. This page explains how to use these generic component
+inside, by wrapping it directly. Also, you can conditionnally add props to manage when the
+element might be swipable or not. 
+
+You can use the component to interact easily with records, messages, items in lists and much more.
+
+.. note:: Actions are permuted when using right-to-left (RTL) languages.
+
+Props
+~~~~~
+
+.. list-table::
+    :widths: 20 20 60
+    :header-rows: 1
+
+    * - Name 
+      - Type
+      - Description
+    * - `onLeftSwipe`
+      - `Object`
+      - if present, the actionswiper can be swiped to the left
+    * - `onRightSwipe`
+      - `Object`
+      - if present, the actionswiper can be swiped to the right
+
+You can use both ``onLeftSwipe`` and ``onRightSwipe`` props at the same time.
+
+Those ``Object``'s must contain:
+    
+    - ``action``, which is the callable ``Function`` serving as a callback.
+      Once the swipe has been completed in the given direction, that action
+      is performed.
+    - ``icon`` is the icon class to use, usually to represent the action.
+      It must be a ``string``.
+    - ``bgColor`` is the background color, given to decorate the action.
+      can be one of the following `bootstrap contextual color <https://getbootstrap.com/docs/3.3/components/#available-variations>`_ (``danger``,
+      ``info``, ``secondary``, ``success`` or ``warning``).
+
+    Those values must be given to define the behavior and the visual aspect
+    of the swiper. 
+
+Example: Create a basic swipable element
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This first example uses some basic features of the ActionSwiper component. 
+Here, the swipe is enabled in both directions.
+
+.. code-block:: xml
+    
+  <ActionSwiper 
+    onRightSwipe="
+      {
+        action: '() => Delete item',
+        icon: 'fa-delete',
+        bgColor: 'bg-danger',
+      }" 
+    onLeftSwipe="
+      {
+        action: '() => Star item',
+        icon: 'fa-star',
+        bgColor: 'bg-warning',
+      }"
+  >
+    <div>
+      Swipable item
+    </div>
+  </ActionSwiper>
+
+Example: Extending existing components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example show you how you can use ``xpath``'s to wrap an existing
+element in the ActionSwiper component. Here, a swiper has been added to mark
+a message as read in mail.
+
+.. code-block:: xml
+    
+  <xpath expr="//*[hasclass('o_Message')]" position="after">
+    <ActionSwiper
+      onRightSwipe="messaging.device.isMobile and messageView.message.isNeedaction ?
+        {
+          action: () => messageView.message.markAsRead(),
+          icon: 'fa-check-circle',
+          bgColor: 'bg-success',
+        } : undefined"
+    />
+  </xpath>
+  <xpath expr="//ActionSwiper" position="inside">
+    <xpath expr="//*[hasclass('o_Message')]" position="move"/>
+  </xpath>
+
+Using the same principle, multiple existing elements have been extended to handle
+swipe interactions.
+
+.. image:: ./images/actionswiper.png
```

Comment:
not sure the screenshot really helps. I would change it to a screenshot of a single action swiper, nothing else, and put it in the description section. Then, remove this and the sentence l283

Reply:
I edited the screenshot to only show one simple component. You're right, it's better like that. I like images in documentations, it helps me as a developer to see what it can do (so please keep this image and encourage the practice :) )